akka-javasdk/src/main/scala/akka/javasdk/impl/workflow/ReflectiveWorkflowRouter.scala

```diff
@@ -75,30 +77,29 @@ class ReflectiveWorkflowRouter[S, W <: Workflow[S]](
     }
   }
 
-  private def methodInvokerLookup(commandName: String) =
+  private def methodInvokerLookup(commandName: String, workflowClass: Class[_]) =
     methodInvokers.getOrElse(
       commandName,
-      throw new HandlerNotFoundException(
-        "command",
-        commandName,
-        instanceFactory(workflowContext).getClass,
-        methodInvokers.keySet))
+      throw new HandlerNotFoundException("command", commandName, workflowClass, methodInvokers.keySet))
 
   final def handleCommand(
       userState: Option[SpiWorkflow.State],
       commandName: String,
       command: BytesPayload,
       context: CommandContext,
       timerScheduler: TimerScheduler,
-      deleted: Boolean): CommandResult = {
+      deleted: Boolean,
+      span: Option[Span]): CommandResult = {
 
-    val workflow = instanceFactory(workflowContext)
+    val updatedContext = workflowContext.asInstanceOf[WorkflowContextImpl].withOpenTelemetrySpan(span)
```

Comment:
Not a big thing, but could be worth not doing anything at all when tracing is disabled (span is None) rather than always create a new instance here regardless.